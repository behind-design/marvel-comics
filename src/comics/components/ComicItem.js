import React from 'react';
import { Link } from 'react-router-dom';
import truncatise from 'truncatise';


const truncateOptions = {
  TruncateBy: 'words',
  TruncateLength: 3,
  StripHTML: true,
  Strict: true,
  Suffix: '...',
};

const fullTitle = comic => `${comic.title} #${comic.issueNumber}`;

const ComicItem = ({ comic }) => (
  <div className="marvel-comics-comic-item" title={fullTitle(comic)}>
    <Link to={`/${comic.id}`}>
      <div className="comic-item-image">
        <img src={comic.thumbnail} alt={comic.title} />
      </div>

      <h3 className="comic-item-title" title={fullTitle(comic)}>{truncatise(fullTitle(comic), truncateOptions)}</h3>
      <div className="comic-item-meta">
        { comic.year && <p>{comic.year}</p> }
        { comic.price && <p>{comic.price}</p> }
      </div>
    </Link>
  </div>
);

export default ComicItem;
