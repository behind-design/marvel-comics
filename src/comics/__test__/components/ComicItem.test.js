import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { mount, render } from 'enzyme';

import ComicItem from '../../components/ComicItem';


const IMAGE_NOT_FOUND = `http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg`;

const comic = JSON.parse(`{"id":16961,"digitalId":0,"title":"Rawhide Kid","issueNumber":134,"variantDescription":"","description":null,"modified":"-0001-11-30T00:00:00-0500","isbn":"","upc":"","diamondCode":"","ean":"","issn":"","format":"Comic","pageCount":0,"textObjects":[],"resourceURI":"http://gateway.marvel.com/v1/public/comics/16961","urls":[{"type":"detail","url":"http://marvel.com/comics/issue/16961/rawhide_kid_1960_134?utm_campaign=apiRef&utm_source=a1af9305675ac96b2505d95b4d442607"}],"series":{"resourceURI":"http://gateway.marvel.com/v1/public/series/2988","name":"Rawhide Kid (1960 - 1979)"},"variants":[],"collections":[],"collectedIssues":[],"dates":[{"type":"onsaleDate","date":"-0001-11-30T00:00:00-0500"},{"type":"focDate","date":"-0001-11-30T00:00:00-0500"}],"prices":[{"type":"printPrice","price":0}],"thumbnail":"http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg","images":[],"creators":{"available":0,"collectionURI":"http://gateway.marvel.com/v1/public/comics/16961/creators","items":[],"returned":0},"characters":{"available":0,"collectionURI":"http://gateway.marvel.com/v1/public/comics/16961/characters","items":[],"returned":0},"stories":{"available":2,"collectionURI":"http://gateway.marvel.com/v1/public/comics/16961/stories","items":[{"resourceURI":"http://gateway.marvel.com/v1/public/stories/35583","name":"","type":""},{"resourceURI":"http://gateway.marvel.com/v1/public/stories/67628","name":"Rawhide Kid 134 cover","type":"cover"}],"returned":2},"events":{"available":0,"collectionURI":"http://gateway.marvel.com/v1/public/comics/16961/events","items":[],"returned":0},"year":"1960","price":"0$"}`);

describe('ComicItem', () => {
  it('renders without crashing', () => {
    mount(
      <MemoryRouter>
        <ComicItem comic={{}} />
      </MemoryRouter>,
    );
  });

  it('renders all the correct properties', () => {
    const wrapper = render(
      <MemoryRouter>
        <ComicItem comic={{
          ...comic,
          title: 'Rawhide Kid',
          year: '1960 - 1979',
        }} />
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
