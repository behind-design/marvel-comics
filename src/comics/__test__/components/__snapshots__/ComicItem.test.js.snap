// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicItem renders all the correct properties 1`] = `
<div
  class="marvel-comics-comic-item"
  title="Rawhide Kid #134"
>
  <a
    href="/16961"
  >
    <div
      class="comic-item-image"
    >
      <img
        alt="Rawhide Kid"
        src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg"
      >
        
      </img>
    </div>
    <h3
      class="comic-item-title"
      title="Rawhide Kid #134"
    >
      Rawhide Kid #134
    </h3>
    <div
      class="comic-item-meta"
    >
      <p>
        1960 - 1979
      </p>
      <p>
        0$
      </p>
    </div>
  </a>
</div>
`;
